tests/unit-tests/00-check/check-eqvp.scm

```diff
@@ -12,7 +12,7 @@
 
 ;; Equivalent NaNs should be eqv?
 (let ((x (fl/ 0. 0.)))
-  (eqv? x x))
+  (check-eqv? x x))
```

Comment:
I don't expect the NaN to be negative. In your example, I'm worried the compiler will do Common Subexpression Elimination (CSE) and convert that back to your original case.  So how about:

```
 (let* ((x (fl/ 0.0 0.0)) (y (fl+ x x))) (check-eqv? x y))
```